src/rewriter.coffee

```diff
@@ -369,6 +371,12 @@ exports.Rewriter = class Rewriter
           break unless isImplicit stackItem
           stackItem[2].sameLine = no if isImplicitObject stackItem
 
+      # End indented-continuation-line implicit objects once that indentation is over.
+      if tag is 'TERMINATOR' and token.endsContinuationLineIndentation
```

Comment:
...the hard part was to then parse additional properties of the outer object as belonging to the outer (not inner) object, eg:
```
a: b or
  b: 1
c: 2 # this belongs to the top-level object, not the nested b: 1 object
```

In order to accomplish that, I added extra token data for "indented line continuations" like this to allow comparing the indentation levels to know when the nested line continuation indent is done (thus when the nested object is done)